taskvine/src/manager/vine_file_replica_table.c

```diff
@@ -36,24 +36,34 @@ int vine_file_replica_table_insert(struct vine_manager *m, struct vine_worker_in
 	return 1;
 }
 
-// remove a file from the remote file table.
-struct vine_file_replica *vine_file_replica_table_remove(
-		struct vine_manager *m, struct vine_worker_info *w, const char *cachename)
+// remove a file from the file worker table.
+int vine_file_worker_table_remove(struct vine_manager *m, struct vine_worker_info *w, const char *cachename)
 {
-	struct vine_file_replica *replica = hash_table_remove(w->current_files, cachename);
-	if (replica) {
-		w->inuse_cache -= replica->size;
-	}
-
 	struct set *workers = hash_table_lookup(m->file_worker_table, cachename);
+
 	if (workers) {
 		set_remove(workers, w);
 		if (set_size(workers) < 1) {
 			hash_table_remove(m->file_worker_table, cachename);
 			set_delete(workers);
+			return 0;
 		}
 	}
 
+	return workers ? set_size(workers) : 0;
```

Comment:
I don't recommend doing this. The returned value should correspond to what the function does. In this case, returning the number of replicas that still exists does not make sense for the function. I'd leave `vine_file_worker_table_remove` as it was, remove `vine_file_worker_table_remove` as removing a replica should always update the corresponding worker and there is no sense of having a second function.

Then in `recover_temp_files` delete the replica and check whether there are still replicas.